js/app.js

```diff
@@ -0,0 +1,64 @@
+
+// Declaro variable para solicitar el número de tarjeta desde un prompt
+var card = prompt('Ingrese número de Tarjeta de Credito a Validar:');
+// Evalua si card esta vacio o está ingresando letras
+if (card === '' || isNaN(card)) {
+// de ser true, brinda un mensaje
+  alert('Esta ingresando campo vacio o letras, Ingrese los datos');
+} else { // caso contrario llama a la funcion isValidCard
+  document.write(isValidCard(card));
+}
+// Funcion para validar el # de una tarjeta de crédito
+function isValidCard(num) {
+// Declaro un arreglo que contendra todos los elementos del número ingresado
+  var arr = [];
+  // Declaro variable que almacenara la validez de la tarjeta
+  var message = '';
+  // recorro el texto ingresado
+  for (var i = 0;i < num.length;i++) {
+  // Declaro variable que almacena el # de la tarjeta ingresada
+    var element = parseInt(num.charAt(i));
+    // agrego el elemento al array creado
+    arr.push(element);
+  }
+  // Declaro un nuevo array que contenga los números en orden invertido
+  var newArr = arr.reverse();
+  // console.log(newArr);
+  // Recorro el arreglo posicion por posicion
+  for (var j = 1;j <= newArr.length;j++) {
+  // evalua si la posicion es par
+    if (j % 2 === 0) {
+      // Declaro variable que almacena el doble del numero
+      var doublePair = newArr[j - 1] * 2;
+      // Evaluo si el doble de ese número es mayor o igual a 10
+      if (doublePair >= 10) {
+      // Separamos los digitos del resultado
+        var digitOne = parseInt(doublePair / 10);
+        var digitTwo = doublePair % 10;
+        // sumamos los  digitos del resultado
+        var sumDigit = digitOne + digitTwo;
+        // Reemplaza en cada posicion par los nuevos numeros
+        newArr[j - 1] = sumDigit;
+      } else {
+        // reemplazamos en cada posicion por el doble del elemento
+        newArr [j - 1] = doublePair;
+      }
+    }
+  }
```

Comment:
El ciclo for puede ser un poco confuso porque la variable `j` empieza en `1` y cada operación se refiere al elemento en el índice anterior `newArr[j - 1]`. En ocasiones es más fácil seguir el flujo de un ciclo cuando el contenido hace referencia al elemento actual, para lograrlo en este ejemplo podrías declararlo así:

```js
for (var j = 0; j < newArr.length; j++) {
  if (j % 2 !== 0) { // diferente de 0
    var doublePair = newArr[j] * 2; // elemento actual
    // resto del código
    } else {
      newArr[j] = doublePair;
    }
  }
}
```